torch/nested/_internal/nested_int.py

```diff
@@ -34,14 +36,32 @@ def _ge(lhs: Any, rhs: Any) -> bool:
         raise ValueError("inputs unsupported")
 
 
+def _get_tensor_id(t: torch.Tensor) -> int:
+    ret: List[Optional[int]] = [None]
+
+    def func(t: torch.Tensor) -> None:
+        if try_get_int(t) is None:
+            ret[0] = register_tensor(t)
+        else:
+            ret[0] = try_get_int(t)
+
+    apply_func(t, func, only_source_fields=True)
+    assert ret[0] is not None
+    return ret[0]
+
+
 class NestedIntNode:
-    def __init__(self, t_id: int, coeff: int):
-        self.t_id = t_id
+    def __init__(self, cache: torch.Tensor, coeff: int):
+        self.cache = cache
+        self.t_id = _get_tensor_id(cache)
         self.coeff = coeff
 
     def nested_int_coeff(self) -> int:
         return self.coeff
 
+    def nested_int_cache(self) -> Any:
+        return self.cache
```

Comment:
small nit, but why prefix the names of `nested_int_cache` / `nested_int_coeff` with `nested_int_` ? feels unnecessary

Reply:
Oh these are here for historic reasons - At one point, these methods also existed on the generic SymNode in cpp and were overridden by the cpp NestedInt which subclassed the base cpp SymNode, and so they needed to be nested specific to indicate what they were for. With nested int in Python I would agree with you that the prefix should be unnecessary.